osu.Game/Screens/Play/HUD/SoloGameplayLeaderboard.cs

```diff
@@ -79,19 +72,16 @@ private void showScores()
             {
                 var score = Add(s.User, false);
 
-                var bindableTotal = scoreManager.GetBindableTotalScore(s);
-
-                // Direct binding not possible due to differing types (see https://github.com/ppy/osu/issues/20298).
-                bindableTotal.BindValueChanged(total => score.TotalScore.Value = total.NewValue, true);
-                scoreBindables.Add(bindableTotal);
-
+                score.GetDisplayScore = s.GetDisplayScore;
+                score.TotalScore.Value = s.TotalScore;
```

Comment:
I'm not sure I understand this. In this case, is there any reason to provide a `TotalScore.Value` update here? Internally, this triggers a `ValueChanged` which causes `updateScore` to get called, but this feels super haphazard. Like I'd hope that setting a new `GetDisplayScore` function did this instead.

And maybe the flow should be changed so the result of `GetDisplayScore` is stored to `TotalScore`, or something similar.

Reply:
I think you need to try out your proposal itself, because it's not easy to make work.

Reply:
Figured as much. I'll keep that in mind when reviewing further.

Reply:
I've at least fixed the part where if these two lines were reversed it would fail to update the initial display score. Probably fine for now.